agent/src/beerocks/slave/tasks/task.h

```diff
@@ -0,0 +1,70 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * SPDX-FileCopyrightText: 2016-2020 the prplMesh contributors (see AUTHORS.md)
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#ifndef _TASK_H_
+#define _TASK_H_
+
+#include <beerocks/tlvf/beerocks_header.h>
+#include <tlvf/CmduMessageRx.h>
+
+namespace beerocks {
+
+class Task {
+public:
+    Task() {}
+    virtual ~Task() {}
+
+    /**
+     * @brief Enum which defines all tasks type.
+     * 
+     * Each task must have an entry on the enum eType, so it will be possible to send the task an
+     * event. 
+     */
+    enum eType : uint8_t {
+        TOPOLOGY,
+    };
+
+    /**
+     * @brief Retrun the task type.
+     * 
+     * Called by TaskPool send_event().
+     * 
+     * @return eType Task type.
+     */
+    virtual eType get_task_type() = 0;
+
+    /**
+     * @brief Work function. Shall do the task routine. Called by TaskPool run_task() function.
+     */
+    virtual void work(){};
```

Comment:
Do we expect any derived `Task` to have _correct_  empty `work()` implementation?
If yes, then fine, the default implementation is correctly set as empty.
If not, and it is illegal to have an empty `work` - please state it clearly -
```suggestion
    virtual void work() = 0;
```

Reply:
We don't.
It is possible that a task would not have work functions. Same as handling CMDUs or events.
For example, a task that only responsible for handling CMDUs.